src/java/org/apache/cassandra/db/compaction/unified/AdaptiveController.java

```diff
@@ -184,9 +225,35 @@ static Controller fromOptions(Environment env,
                                       targetSSTableSize,
                                       overlapInclusionMethod,
                                       intervalSec,
-                                      minScalingParameter, maxScalingParameter,
+                                      minScalingParameter,
+                                      maxScalingParameter,
                                       threshold,
-                                      minCost, maxAdaptiveCompactions);
+                                      minCost,
+                                      maxAdaptiveCompactions,
+                                      keyspaceName,
+                                      tableName);
+    }
+
+    private static int[] readStoredScalingParameters(JSONArray storedScalingParameters)
+    {
+        if (storedScalingParameters.size() > 0)
+        {
+            int[] scalingParameters = new int[UnifiedCompactionStrategy.MAX_LEVELS];
+            for (int i = 0; i < scalingParameters.length; i++)
+            {
+                //if the file does not have enough entries, use the last entry for the rest of the levels
```

Comment:
I still feel like this will cause confusion for operators because it's implicit rather than explicit. If I haven't set a value for a level in a file, I would expect it to be the default value, not the same as the last value that I set.

My preference would be to use some default here, even if the default is "unset". Is there something I'm not seeing? @ethan-brown2022 @blambov ?

Reply:
This is generally how it acts in other areas, so I think it makes sense to keep the same behavior here.  The only instance I can think of where the file would contain some stored scaling parameters but not one for every level is when switching from static to adaptive.  If an operator is using a static configuration and gave `'scaling_parameters' : '4, 0, -2, -4'`, that is the same as saying `'scaling_parameters': '4, 0, -2, -4, -4, -4, ...'`.  If an operator was using this static configuration and switched to adaptive, I imagine they would expect the scaling parameters to start as [4, 0, -2, -4, -4, -4, ...] instead of becoming [4, 0, -2, -4, default, default, ...].